tuf/ngclient/fetcher.py

```diff
@@ -28,38 +28,65 @@ class FetcherInterface:
     __metaclass__ = abc.ABCMeta
 
     @abc.abstractmethod
-    def fetch(self, url: str) -> Iterator[bytes]:
+    def _fetch(self, url: str) -> Iterator[bytes]:
         """Fetches the contents of HTTP/HTTPS url from a remote server.
 
+        Implementations must raise DownloadHTTPError if they receive an
+        HTTP error code.
+
+        Implementations may raise any errors but the ones that are not
+        DownloadErrors will be wrapped in a DownloadError by fetch().
+
         Arguments:
             url: A URL string that represents a file location.
 
         Raises:
-            exceptions.SlowRetrievalError: A timeout occurs while receiving
-                data.
-            exceptions.FetcherHTTPError: An HTTP error code is received.
+            exceptions.DownloadHTTPError: An HTTP error code was received.
 
         Returns:
             A bytes iterator
         """
         raise NotImplementedError  # pragma: no cover
 
+    def fetch(self, url: str) -> Iterator[bytes]:
+        """Fetches the contents of HTTP/HTTPS url from a remote server.
+
+        Arguments:
+            url: A URL string that represents a file location.
+
+        Raises:
+            exceptions.DownloadError: An error occurred during download.
+            exceptions.DownloadHTTPError: An HTTP error code was received.
+
+        Returns:
+            A bytes iterator
+        """
+        # Ensure that fetch() only raises DownloadErrors, regardless of the
+        # fetcher implementation
+        try:
+            return self._fetch(url)
+        except Exception as e:
+            if isinstance(e, exceptions.DownloadError):
+                raise e
+            raise exceptions.DownloadError(f"Failed to download {url}") from e
```

Comment:
Probably a matter of taste, but wouldn't this look more idiomatic?

```suggestion
        except exceptions.DownloadError:
            raise
        except Exception as e
            raise exceptions.DownloadError(f"Failed to download {url}") from e
```

Reply:
oh definitely, I'll change this